friends/src/components/Friends.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import './Friends.css';
+
+class Friends extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      friends: []
+    };
+  }
+
+  deleteFriend = (e, id) => {
+    e.preventDefault();
+    this.props.deleteFriend(id);
+  };
+
+  updateForm = (e, id) => {
+    e.preventDefault();
+    this.props.setUpdateForm(id);
+  };
+
+  render() {
+    return (
+      <div>
+        <h1>Friends List</h1>
+        {this.props.friends.map(friend => (
+          <div>
+            <p className="name"><strong>Name: {friend.name}</strong></p>
+            <p>Age: {friend.age}</p>
+            <p>Email: {friend.email}</p>
+            <button type="submit" onClick={(e) => this.deleteFriend(e, friend.id)}><strong>Delete Friend</strong></button>
+            <button type="submit" onClick={(e) => this.updateForm(e, friend.id)}><strong>UpdateFriend</strong></button>
```

Comment:
Here's where we pass in friend.id. Start on line 32 and pass in the whole `friend` object instead of `friend.id`. Refactor the rest of your code to use that friend object and make your update functionality work.